src/us/kbase/workspace/database/provenance/ContributorRole.java

```diff
@@ -0,0 +1,113 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.HashMap;
+import java.util.Map;
+import us.kbase.workspace.database.Util;
+
+/**
+ * A class representing contributor roles
+ */
+public enum ContributorRole {
+
+	// DataCite contributor roles
+	CONTACT_PERSON("DataCite", "ContactPerson"),
+	DATA_COLLECTOR("DataCite", "DataCollector"),
+	DATA_CURATOR("DataCite", "DataCurator"),
+	DATA_MANAGER("DataCite", "DataManager"),
+	DISTRIBUTOR("DataCite", "Distributor"),
+	EDITOR("DataCite", "Editor"),
+	HOSTING_INSTITUTION("DataCite", "HostingInstitution"),
+	PRODUCER("DataCite", "Producer"),
+	PROJECT_LEADER("DataCite", "ProjectLeader"),
+	PROJECT_MANAGER("DataCite", "ProjectManager"),
+	PROJECT_MEMBER("DataCite", "ProjectMember"),
+	REGISTRATION_AGENCY("DataCite", "RegistrationAgency"),
+	REGISTRATION_AUTHORITY("DataCite", "RegistrationAuthority"),
+	RELATED_PERSON("DataCite", "RelatedPerson"),
+	RESEARCHER("DataCite", "Researcher"),
+	RESEARCH_GROUP("DataCite", "ResearchGroup"),
+	RIGHTS_HOLDER("DataCite", "RightsHolder"),
+	SPONSOR("DataCite", "Sponsor"),
+	SUPERVISOR("DataCite", "Supervisor"),
+	WORK_PACKAGE_LEADER("DataCite", "WorkPackageLeader"),
+	OTHER("DataCite", "Other"),
+	// CRediT contributor roles
+	CONCEPTUALIZATION("CRediT", "conceptualization"),
+	DATA_CURATION("CRediT", "data-curation"),
+	FORMAL_ANALYSIS("CRediT", "formal-analysis"),
+	FUNDING_ACQUISITION("CRediT", "funding-acquisition"),
+	INVESTIGATION("CRediT", "investigation"),
+	METHODOLOGY("CRediT", "methodology"),
+	PROJECT_ADMINISTRATION("CRediT", "project-administration"),
+	RESOURCES("CRediT", "resources"),
+	SOFTWARE("CRediT", "software"),
+	SUPERVISION("CRediT", "supervision"),
+	VALIDATION("CRediT", "validation"),
+	VISUALIZATION("CRediT", "visualization"),
+	WRITING_ORIGINAL_DRAFT("CRediT", "writing-original-draft"),
+	WRITING_REVIEW_EDITING("CRediT", "writing-review-editing");
+
+	// mapping of various name formats to ContributorRole
+	private static final Map<String, ContributorRole> STRING_TO_ROLE_MAP = new HashMap<>();
+	static {
+		for (final ContributorRole cr : ContributorRole.values()) {
+			STRING_TO_ROLE_MAP.put(cr.name().toLowerCase(), cr);
+			STRING_TO_ROLE_MAP.put(cr.name().replace("_", "").toLowerCase(), cr);
+			STRING_TO_ROLE_MAP.put(cr.getSource().toLowerCase() + ":" + cr.name().replace("_", "").toLowerCase(), cr);
+			STRING_TO_ROLE_MAP.put(cr.getIdentifier().toLowerCase(), cr);
+			STRING_TO_ROLE_MAP.put(cr.getPid().toLowerCase(), cr);
+			STRING_TO_ROLE_MAP.put(cr.getSource().toLowerCase() + ":" + cr.name().toLowerCase(), cr);
```

Comment:
some key overwriting here as some of the transformations produce the same strings